VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,216 @@
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, label, creationDate), now,
+                answerPrefix + " " + relativeTime + " " + outputAuthor + " " + outputAuthorLbl);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, creationDate), now,
+                answerPrefix + " " + relativeTime + " " + outputAuthorLbl);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, creationDate), now,
+                answerPrefix + " " + relativeTime + " " + outputAuthor);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, label, null), now,
+                answerPrefix + " " + outputAuthor + " " + outputAuthorLbl);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, creationDate), now,
+                answerPrefix + " " + relativeTime);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, null), now,
+                answerPrefix + " " + outputAuthor);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, null), now,
+                answerPrefix + " " + outputAuthorLbl);
+
+        // Empty Case
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, null), now,
+                null);
+    }
+
+    private void assertSetAuthorAttributionText(TextView textView,
+                                                DiscussionTextUtils.AuthorAttributionLabel type,
+                                                IAuthorData input,
+                                                long now, String expectedOutput) {
+        final Runnable listener = Mockito.mock(Runnable.class);
+        DiscussionTextUtils.setAuthorAttributionText(textView, type, input, now, listener);
+        if (expectedOutput == null) {
+            assertTrue(textView.getVisibility() == View.GONE);
+        } else {
+            String output = textView.getText().toString();
+            assertEquals(expectedOutput, output);
+            if (!input.isAuthorAnonymous()) {
+                // Test whether author span is clickable or not
+                int start = output.indexOf(input.getAuthor());
+                int end = start + input.getAuthor().length();
+                SpannedString text = (SpannedString) textView.getText();
```

Comment:
Instead of casting to a specific implementation of the `Spanned` interface here (`SpannedString`), you should just be casting it to the interface itself I guess.